### PR TITLE
Refactor readlimit_test.go to appease CI.

### PR DESCRIPTION
**Pull request by lthibault**
Attempting to kill three birds with one stone:

1. (Primary) Fix spurious CI failures (apparently triggered by `t.Log`)
2. Progressively migrate unit tests to `github.com/stretchr/testify`
3. Run tests in parallel

### CODE REVIEW

File: readlimit_test.go [hunk 1]
```diff
@@ -65,113 +71,64 @@ func TestMessage_canRead(t *testing.T) {
...
-		t.Log("   m := new(Message)")
-		m.ResetReadLimit(0)
-		t.Log("   m.ResetReadLimit(0)")
-		if m.canRead(1) {
-			t.Error("!! m.canRead(1) -> true; want false")
-		} else {
-			t.Log("   m.canRead(1) -> false")
-		}
+	t.Parallel()
+	t.Helper()
```
Comment (zenhack) [inline readlimit_test.go:75]: Why are we using Helper() here?

**Review by zenhack — COMMENTED** [review]
One minor question.

File: readlimit_test.go (hunk 1 above)
Comment (lthibault) [reply readlimit_test.go:75]: All the tests are being run as subtests (see line 97).  This helps keep the output tidy.

Comment (zenhack) [reply readlimit_test.go:75]: Fair enough. lgtm, but you've got this marked as a draft.

**Comment by lthibault** [comment]
This is turning into a bit of a yak shave, so I'm ready to just merge this in as-is.  Nothing has changed since your last review, but feel free to give it another look if you want.  Else, feel free to merge.

**Review by zenhack — APPROVED** [review]

**Pull request merged**